### DISCUSSION TITLE
4.0.0-canary.2

### DISCUSSION THREAD

**Discussion by LabhanshAgrawal** [vercel/hyper#6968] [Announcements]
Custom config migration

This release enables migration of any custom logic if present from the v3 config to a local plugin `migrated-hyper3-config`. If you're upgrading from 4.0.0-canary.1 then you'd need to delete the previously generated config file for migration to take place. The local plugins are located at `~/.config/Hyper/plugins/local` for mac & linux, and to `%APPDATA%\Hyper\plugins\local` for windows. It should be in `$XDG_CONFIG_HOME/Hyper/plugins/local` if `$XDG_CONFIG_HOME` is set.

The window border radius on macOS is also fixed now.

I hope you all will give this release a try and post your old config vs the generated plugin in the release discussions thread and report any bugs.

|   OS    | Installer |                                                                                                                   |                                                                                                                    |
| :-----: | :-------: | :---------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------------------------: |
|   Mac   |    dmg    | [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/Hyper-4.0.0-canary.2-mac-x64.dmg) (Intel) | [arm64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/Hyper-4.0.0-canary.2-mac-arm64.dmg) (M1) |
|  Linux  |    deb    |      [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/hyper_4.0.0-canary.2_amd64.deb)      |     [arm64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/hyper_4.0.0-canary.2_arm64.deb)      |
|         |    rpm    |     [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/hyper-4.0.0-canary.2.x86_64.rpm)      |    [arm64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/hyper-4.0.0-canary.2.aarch64.rpm)     |
|         | AppImage  |      [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/Hyper-4.0.0-canary.2.AppImage)       |   [arm64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/Hyper-4.0.0-canary.2-arm64.AppImage)   |
|         |   snap    |     [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/hyper_4.0.0-canary.2_amd64.snap)      |                                                                                                                    |
| Windows |    exe    |      [x64](https://github.com/vercel/hyper/releases/download/v4.0.0-canary.2/Hyper-Setup-4.0.0-canary.2.exe)      |                                                                                                                    |

## Changelog
- Update window border radius in f494781ff22171cb4ab6081dbd6f71b13ae5f9a3
- Migrate custom v3 config to local plugins if needed in https://github.com/vercel/hyper/pull/6967

**Note** - Set `updateChannel` to `'canary'` in hyper config (`command/ctrl`+`,`) to stay on canary builds.

**Full Changelog**: https://github.com/vercel/hyper/compare/v4.0.0-canary.1...v4.0.0-canary.2

<hr /><em>This discussion was created from the release <a href='https://github.com/vercel/hyper/releases/tag/v4.0.0-canary.2'>4.0.0-canary.2</a>.</em>

**Comment by TimeTerminal** [comment]
Hey @LabhanshAgrawal just wanted to thank you for your continued updates to Hyper. It seems like the terminal isn't receiving that many big updates recently, but appreciate these small ones nonetheless.

**Comment by JJJ** [comment]
Both canary.1 and canary.2 working as intended for me.

* macOS 13.2 Beta (M1)
  * Border radius fix looks good!
* Plugins moved to `.config/Hyper`
* `hyper.json` created
  * included all of my custom keys & values
  * didn't see `migrated-hyper3-config`

**Reply by LabhanshAgrawal** [reply]
The local plugin will only be created if you have something in your config which can't be serialized to JSON like functions etc

**Reply by JJJ** [reply]
> The local plugin will only be created if you have something in your config which can't be serialized to JSON like functions etc

Understood! Just wanted to confirm it was correctly not created. 👍
